packages/grafana-ui/src/components/InfoBox/InfoBox.tsx

```diff
@@ -24,24 +27,61 @@ export interface InfoBoxProps extends Omit<React.HTMLAttributes<HTMLDivElement>,
   severity?: AlertVariant;
   /** Call back to be performed when box is dismissed */
   onDismiss?: () => void;
+  /** If dismissPersistenceId is set, the info box will be peristently dismissed when the user dismisses the infobox.*/
+  dismissPersistenceId?: string;
 }
 
 /**
   @public
  */
 export const InfoBox = React.memo(
   React.forwardRef<HTMLDivElement, InfoBoxProps>(
-    ({ title, className, children, branded, url, urlTitle, onDismiss, severity = 'info', ...otherProps }, ref) => {
+    (
+      {
+        title,
+        className,
+        children,
+        branded,
+        url,
+        urlTitle,
+        onDismiss,
+        dismissPersistenceId,
+        severity = 'info',
+        ...otherProps
+      },
+      ref
+    ) => {
       const theme = useTheme();
       const styles = getInfoBoxStyles(theme, severity);
       const wrapperClassName = branded ? cx(styles.wrapperBranded, className) : cx(styles.wrapper, className);
 
+      let dismiss;
+
+      if (dismissPersistenceId) {
+        const persistanceId = FEATUREINFOBOX_PERSISTENCE_ID_PREFIX.concat(dismissPersistenceId);
+
+        const [dismissed, setDismissed] = useLocalStorage(persistanceId, { isDismissed: false });
```

Comment:
I think this will violate the rules of hooks as `useLocalStorage` is called conditionally here. We have a lint rule for this in the main codebase but it's currently disabled in grafana-ui.

Reply:
Ah, yeah I was having some issues with this approach and useLocalStorage. If dismissPersistenceId isn't set it would create an entry in localstorage with an empty string as the key. I had this suspicion that moving the persistence stuff into a separate component that wraps the info boxes might be a better idea. I think I'll try that out instead.